src/apps/loan-list/list/loan-list.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useState } from "react";
+import dayjs from "dayjs";
+import localizedFormat from "dayjs/plugin/localizedFormat";
+import MenuIcon from "@danskernesdigitalebibliotek/dpl-design-system/build/icons/collection/Menu.svg";
+import VariousIcon from "@danskernesdigitalebibliotek/dpl-design-system/build/icons/collection/Various.svg";
+import { getLoansV2 } from "../../../core/fbs/fbs";
+import { LoanV2 } from "../../../core/fbs/model/loanV2";
+import MaterialDecorator from "../materials/material-decorator";
+import { useText } from "../../../core/utils/text";
+import DueDateLoansModal from "../modal/due-date-loans-modal";
+
+const LoanList: React.FC = () => {
+  const t = useText();
+  const [loans, setLoans] = useState<LoanV2[] | null>([]);
+  const [dueDates, setDueDates] = useState<string[]>();
+  const [dueDateModal, setDueDateModal] = useState<string>("");
+  const [loansModal, setLoansModal] = useState<LoanV2[] | null>();
+  const [renewable, setRenewable] = useState<number | null>(null);
+  const [amountOfLoans, setAmountOfLoans] = useState<number>(0);
+  const [view, setView] = useState<string>("list");
+
+  useEffect(() => {
+    dayjs.extend(localizedFormat);
+  }, []);
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(window.location.search);
+    // regex for finding date string from modal query param
+    const regex = /^\d{4}-\d{2}-\d{2}$/;
+    // modal query param
+    const modalString = searchParams.get("modal");
+
+    if (modalString && loans) {
+      const found = modalString.toString().match(regex);
+      // If there is a date string in the modal query param
+      if (found) {
+        setDueDateModal(found[0]);
+        // THe loans are filtered with said date string
+        const loansForModal = loans.filter(
+          ({ loanDetails }) => loanDetails.dueDate === dueDateModal
+        );
+        // Amount of renewable loans are determined, used in the ui
+        const amountOfRenewableLoans = loansForModal.filter(
+          ({ isRenewable }) => isRenewable
+        ).length;
+
+        // Loans for modal (the modal shows loans stacked by due date)
+        setLoansModal(loansForModal);
+        setRenewable(amountOfRenewableLoans);
+      }
+    }
+  }, [loans, dueDateModal]);
+
+  useEffect(() => {
+    getLoansV2().then((value) => {
```

Comment:
I think we should use the [useGetLoansV2](https://github.com/danskernesdigitalebibliotek/dpl-react/blob/c36e34c7dc23455b18459b23a68a1104364bf535/src/core/fbs/fbs.ts#L1061) hooks here. That is how we are doing in the other places where use one of the API's.
In that way we get a unified way of requesting no matter if we are an Orval based client or graphql-codegen generated client.
The useXxx hooks uses react-query under the hood, which gives as a lot of nice tools we can use.
Here is an example using another service from fbs:
https://github.com/danskernesdigitalebibliotek/dpl-react/blob/release/3/src/components/availability-label/availability-label.tsx#L22

Reply:
I honestly have no idea why i didnt do that to begin with